app/routes.py

```diff
@@ -1,2 +1,333 @@
-from flask import Blueprint
+import re
+from flask import Blueprint, jsonify, request
+from flask.helpers import make_response
+from app.models.task import Task
+from app.models.goal import Goal
+import requests
+from datetime import datetime 
+from app import db
+import os
+from dotenv import load_dotenv
 
+tasks_bp = Blueprint("tasks_bp", __name__, url_prefix="/tasks")
+
+@tasks_bp.route("", methods=["POST"])
+def create_task():
+    request_body = request.get_json()
+    
+    if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
+        return jsonify({"details": "Invalid data"}), 400
+    
+    new_task = Task(
+        title=request_body["title"], 
+        description=request_body["description"], 
+        completed_at=request_body["completed_at"]
+        )
+    
+    db.session.add(new_task) 
+    db.session.commit()
+    
+    response_body = {
+        "task": {
+            "id": new_task.task_id,
+            "title": new_task.title,
+            "description": new_task.description,
+            "is_complete": new_task.completed_at is not None
+            }
+        }
```

Comment:
We can refactor this function by moving the dictionary logic into a helper function for the `Task` model. 

```suggestion
    response_body = {
        "task": task.to_dict()
        }
```